server/src/main/java/org/elasticsearch/action/main/TransportMainAction.java

```diff
@@ -49,8 +50,18 @@ protected void doExecute(Task task, MainRequest request, ActionListener<MainResp
         ClusterState clusterState = clusterService.state();
         assert Node.NODE_NAME_SETTING.exists(settings);
         final boolean available = clusterState.getBlocks().hasGlobalBlock(RestStatus.SERVICE_UNAVAILABLE) == false;
+
+        // If we cannot find the cluster dynamic name in our dynamic settings, we must fallback
```

Comment:
I don't think we want to do this? it's a property cluster people can set if they want. Doing this like this opens the door for people forgetting this on some nodes and then it is sometimes there as sometimes not?

Reply:
I'm not sure I follow you here -- the `cluster.name` attribute must be the same on all nodes in the cluster for nodes to be able to join the cluster, correct?  The issue here is that `cluster.name` only appears in the node-level settings object, not in the one returned from the cluster metadata, so we have to look at both.

Reply:
`cluster.name` is special because it has to be there and you can't forget it (i.e., if you do, the default works for you and then it's set). If we're after the fallback to a cluster name (barring other discussion) we need to only look for the cluster name setting in the node level settings and the dynamic name setting (with a fallback). The way the code is now allows people to set the dynamic name in the node settings and it will work. Does that make sense?

Reply:
Yes that makes sense, thanks!  Will hold off on this change pending resolution of the fallback discussion.